common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -101,9 +104,10 @@ public Object run() {
             // is an instanceof Unsafe and reversing the if and else blocks; this is because an
             // instanceof check against Unsafe will trigger a class load and we might not have
             // the runtime permission accessClassInPackage.sun.misc
-            if (maybeUnsafe instanceof Exception) {
+            if (maybeUnsafe instanceof Throwable) {
                 unsafe = null;
-                logger.debug("sun.misc.Unsafe.theUnsafe: unavailable", (Exception) maybeUnsafe);
+                unsafeUnavailabilityCause = (Throwable) maybeUnsafe;
+                logger.debug("sun.misc.Unsafe.theUnsafe: unavailable", (Throwable) maybeUnsafe);
             } else {
                 unsafe = (Unsafe) maybeUnsafe;
                 logger.debug("sun.misc.Unsafe.theUnsafe: available");
```

Comment:
there are a few more cases below where it would be useful to set the `unsafeUnavailabilityCause` (where ever `unsafe` is set to `null`.

Reply:
+1